src/util.rs

```diff
@@ -109,8 +114,8 @@ pub fn time_to_nstime(t: &time::OffsetDateTime) -> i64 {
     t.unix_timestamp()
 }
 
-/// Utility function safely converting a slice of `i8` values into a `String`.
-pub(crate) fn i8_to_string(buf: &[i8]) -> String {
+/// Utility function safely converting a slice of `c_char` values into a `String`.
+pub(crate) fn to_string(buf: &[c_char]) -> String {
     let v: Vec<u8> = buf
         .iter()
         .map(|x| *x as u8) // cast i8 as u8
```

Comment:
Should `i8` in this comment have been changed to `c_char`?

Reply:
this is a pub fn and that's why we add `doc comments` as a part of rust code practice.

Reply:
This comment is on the `// cast i8 as u8` line 121.

I believe the `i8` in this comment should have been changed to `c_char` similar to the changes on lines 117&118.

Or removed since the incorrect `i8` in the comment is unnecessary duplication that is causing a problem and the `cast ... as u8` part of the comment is clear in the `as u8` part of the code.

Reply:
Hi @cordahi the owner of the mseed crate mergerd my PR to the main repo and released a new version, so should we continue this PR review? what do you think?

Reply:
1. This PR can be closed and this branch deleted and your main branch fast-forwarded to match the damb main branch.
2. This discussion is IMO useful, to me it indicates that the frequent casts required by Rust which are implicit "unsafe" make Rust not as safe a language as it makes itself out to be.
3. Some of this can be proposed as a follow-up to the PR that has been merged in original repository, e.g. this missed change resulting in an incorrect comment.